mobilitypath_publisher/src/mobilitypath_publisher.cpp

```diff
@@ -13,137 +13,182 @@
+  {
+    carma_v2x_msgs::msg::MobilityPath mobility_path_msg;
+    uint64_t millisecs = (trajectory_plan.header.stamp.sec*1000000000 + trajectory_plan.header.stamp.nanosec) / 1000000;
+    mobility_path_msg.m_header = composeMobilityHeader(millisecs);
         
-
-        mob_path_pub_.publish(latest_mobility_path_); 
-        return true;
+    if (!map_projector_) {
+      RCLCPP_ERROR_STREAM(this->get_logger(), "MobilityPath cannot be populated as map projection is not available");
+      return mobility_path_msg;
     }
 
-    void MobilityPathPublication::georeference_cb(const std_msgs::StringConstPtr& msg) 
-    {
-        map_projector_ = std::make_shared<lanelet::projection::LocalFrameProjector>(msg->data.c_str());  // Build projector from proj string
-    }
+    carma_v2x_msgs::msg::Trajectory mob_path_traj = TrajectoryPlantoTrajectory(trajectory_plan.trajectory_points);
+    mobility_path_msg.trajectory = mob_path_traj;
 
-    void MobilityPathPublication::trajectory_cb(const cav_msgs::TrajectoryPlanConstPtr& msg)
-    {
-        latest_trajectory_ = *msg;
+    return mobility_path_msg;
+  }
 
-        latest_mobility_path_ = mobilityPathMessageGenerator(latest_trajectory_);
-    }
+  carma_v2x_msgs::msg::MobilityHeader MobilityPathPublication::composeMobilityHeader(uint64_t time)
+  {
+    carma_v2x_msgs::msg::MobilityHeader header;
+    header.sender_id = config_.vehicle_id;
+    header.recipient_id = recipient_id;
+    header.sender_bsm_id = bsmIDtoString(bsm_core_);
 
-    void MobilityPathPublication::bsm_cb(const cav_msgs::BSMConstPtr& msg)
-    {
-        bsm_core_ = msg->core_data;
-        
-    }
-    
-// @SONAR_START@
-    cav_msgs::MobilityPath MobilityPathPublication::mobilityPathMessageGenerator(const cav_msgs::TrajectoryPlan& trajectory_plan)
-    {
-        cav_msgs::MobilityPath mobility_path_msg;
-        uint64_t millisecs =trajectory_plan.header.stamp.toNSec()/1000000;
-        mobility_path_msg.m_header = composeMobilityHeader(millisecs);
+    // random GUID that identifies this particular plan for future reference
+    header.plan_id = boost::uuids::to_string(boost::uuids::random_generator()());
+    header.timestamp = time; //time in millisecond
         
-        if (!map_projector_) {
-            ROS_ERROR_STREAM("MobilityPath cannot be populated as map projection is not available");
-            return mobility_path_msg;
-        }
+    return header;
+  }
 
-        cav_msgs::Trajectory mob_path_traj = TrajectoryPlantoTrajectory(trajectory_plan.trajectory_points);
-        mobility_path_msg.trajectory = mob_path_traj;
+  carma_v2x_msgs::msg::Trajectory MobilityPathPublication::TrajectoryPlantoTrajectory(const std::vector<carma_planning_msgs::msg::TrajectoryPlanPoint>& traj_points) const
+  {
+    carma_v2x_msgs::msg::Trajectory traj;
 
-        return mobility_path_msg;
-    }
+    carma_v2x_msgs::msg::LocationECEF ecef_location = TrajectoryPointtoECEF(traj_points[0]); //m to cm to fit the msg standard 
```

Comment:
Lets add a check for empty() prior to this line to avoid seg faults if an empty trajectory gets published.